File: ios/NimbleComms.m
//
//  NimbleComms.m
//  nimbleswiftreactdemo
//
//  Created by Kieran Hall on 07/03/2023.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(NimbleComms, NSObject)
RCT_EXTERN_METHOD(getData: (RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(setName: (NSString)name)
@end

File: ios/NimbleComms.swift
//
//  NimbleComms.swift
//  nimbleswiftreactdemo
//
//  Created by Kieran Hall on 07/03/2023.
//

import Foundation

@objc(NimbleComms)
class NimbleComms: NSObject {
  private var name = "Ernie Wise"
  private var fruits = ["apple", "pear", "raspberry", "strawberry", "lemon", "lime"]
  
  @objc
  func getData(_ callback: RCTResponseSenderBlock){
    callback([
      fruits.randomElement() ?? "",
      name
    ])
  }
  
  @objc
  func setName(_ name: NSString){
    self.name = name as String
  }
}
